pkg/api/validation/validation.go

```diff
@@ -1937,6 +1948,50 @@ func ValidateVolumeMounts(mounts []api.VolumeMount, volumes sets.String, contain
 	return allErrs
 }
 
+func ValidateVolumeDevices(devices []api.VolumeDevice, volumes sets.String, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	devicepath := sets.NewString()
+	devicename := sets.NewString()
+
+	if devices != nil && !utilfeature.DefaultFeatureGate.Enabled(features.BlockVolumeSupport) {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("volumeDevices"), "Container volumeDevices is disabled by feature-gate"))
+		return allErrs
+	}
+	if devices != nil {
+		for i, dev := range devices {
+			idxPath := fldPath.Index(i)
+			devName := dev.Name
+			devPath := dev.DevicePath
+			if len(devName) == 0 {
+				allErrs = append(allErrs, field.Required(idxPath.Child("name"), ""))
+			} else if !volumes.Has(devName) {
```

Comment:
Do we need to validate that only persistentVolumeClaims are allowed?

Reply:
hmmm...as opposed to in-line?  Shouldn't that be handled by the plugin?

Reply:
I'm open to adding this if you think we should, so VolumeDevice name must match volumes.name + VolumeSource == PVC

Reply:
I believe as part of the design discussion, if we only allow PVCs, then we do not need any changes to PodSecurityPolicy.

Reply:
OK, I can add that in... Did you see my other question? @msau42 do I need to do anything with the settings apis objects for PodPresetSpec?